plc4j/integrations/apache-nifi/nifi-plc4x-processors/src/main/java/org/apache/plc4x/nifi/util/Plc4xCommon.java

```diff
@@ -258,34 +259,43 @@ else if (value.isTime())
 	 * @param addresses
 	 * @return
 	 */
-	public static Map<String, String> parseAddressString(String connectionString, PropertyValue addresses){
+	public static Map<String, String> parseAddressString(String connectionString, PropertyValue addressesProp){
 		Map<String, String> addressMap = new HashMap<>();
-		//TODO OPCUA var adresses sintax
+		String addresses = addressesProp.getValue();
+		
```

Comment:
This is a big yellow warning sign/code smell however you want to call it.
The insistence on having an option do do something in a way that is more complicated, error prone, and requires this kind of stuff to work is an anti-pattern.

Also, it seems you want to add *more* properties, to help control how this complicated code works, to make it even more complicated.

I will not ask you to change it however.